nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/IdentifyMimeType.java

```diff
@@ -89,14 +89,17 @@
 @Tags({"compression", "gzip", "bzip2", "zip", "MIME", "mime.type", "file", "identify"})
 @CapabilityDescription("Attempts to identify the MIME Type used for a FlowFile. If the MIME Type can be identified, "
         + "an attribute with the name 'mime.type' is added with the value being the MIME Type. If the MIME Type cannot be determined, "
-        + "the value will be set to 'application/octet-stream'. In addition, the attribute mime.extension will be set if a common file "
-        + "extension for the MIME Type is known.")
+        + "the value will be set to 'application/octet-stream'. In addition, the attribute 'mime.extension' will be set if a common file "
+        + "extension for the MIME Type is known. If the MIME Type detected is of type text/*, attempts to identify the charset used " +
+        "and an attribute with the name 'mime.charset' is added with the value being the charset.")
 @WritesAttributes({
-@WritesAttribute(attribute = "mime.type", description = "This Processor sets the FlowFile's mime.type attribute to the detected MIME Type. "
-        + "If unable to detect the MIME Type, the attribute's value will be set to application/octet-stream"),
-@WritesAttribute(attribute = "mime.extension", description = "This Processor sets the FlowFile's mime.extension attribute to the file "
-        + "extension associated with the detected MIME Type. "
-        + "If there is no correlated extension, the attribute's value will be empty")
+        @WritesAttribute(attribute = "mime.type", description = "This Processor sets the FlowFile's mime.type attribute to the detected MIME Type. "
+                + "If unable to detect the MIME Type, the attribute's value will be set to application/octet-stream"),
+        @WritesAttribute(attribute = "mime.extension", description = "This Processor sets the FlowFile's mime.extension attribute to the file "
+                + "extension associated with the detected MIME Type. "
+                + "If there is no correlated extension, the attribute's value will be empty"),
+        @WritesAttribute(attribute = "mime.charset", description = "This Processor sets the FlowFile's mime.charset attribute to the detected charset. "
+                + "If unable to detect the charset or the detected MIME type is not of type text/*, the attribute's value will be empty")
```

Comment:
As noted below, it seems better to avoid setting the attribute instead of setting an empty value for null.

Reply:
Thank you for the feedback. 
I adjusted the implementation and documentation. In the original commit, I followed the implementation of `mime.extension`, which is set to an empty string, when it cannot be determined. Should its implementation as is or don't be set like `mime.charset` as well?

Reply:
Thanks for making the change to the new `mime.charset` approach. I think the existing approach for `mime.extension` should be left unchanged to avoid unexpected changes for existing flows. Since this is a new attribute that only applies for text types, it seems better to set it only when it is not null.

Reply:
Perfect. Yes, that makes sense. Thank you for your quick responses.